### ISSUE TITLE
Deployment fails because functions defined in the notebook are not found

### ISSUE THREAD

**Issue by dmbates** [rikhuijzer/JuliaTutorialsTemplate#6]
First, thanks for providing this resource.  I have tried to use it to create a tutorial in https://github.com/dmbates/Wordlebase3 but the build and deploy step is failing when building the notebooks

https://github.com/dmbates/Wordlebase3/runs/5060532639?check_suite_focus=true

The error is an `undefVarError` for `score`, which is a function with one method that is defined in the notebook itself.

Did I forget to update a configuration file or something like that?

**Comment by rikhuijzer** [comment]
> The error is an undefVarError for score, which is a function with one method that is defined in the notebook itself.

It looks like it is caused by BenchmarkTools. When running the notebook locally in Pluto, I see this error in one of the cells:

![image](https://user-images.githubusercontent.com/20724914/152525765-c71e5539-049f-499b-8426-b019bb9cdbed.png)

This means that this is not an issue due to JuliaTutorialsTemplate or PlutoStaticHTML so I'll close this.

The cryptic error (copy-pasted below) can definitely be improved because I agree that this was more tricky to figure out than necessary.

```
ERROR: UndefVarError: score not defined
Details:
Dict{Symbol, Any}[Dict(:call => "var\"##core#390\"(::String, ::String)", :inlined => false, :file => "execution.jl", :line => 483, :path => "/home/runner/.julia/packages/BenchmarkTools/SnZV6/src/execution.jl"), Dict(:call => "var\"##sample#391\"(::Tuple{}, ::BenchmarkTools.Parameters)", :inlined => false, :file => "execution.jl", :line => 489, :path => "/home/runner/.julia/packages/BenchmarkTools/SnZV6/src/execution.jl"), Dict(:call => "var\"#_run#48\"(::Bool, ::String, ::Base.Pairs{Symbol, Integer, NTuple{4, Symbol}, NamedTuple{(:samples, :evals, :gctrial, :gcsample), Tuple{Int64, Int64, Bool, Bool}}}, ::typeof(BenchmarkTools._run), ::BenchmarkTools.Benchmark, ::BenchmarkTools.Parameters)", :inlined => false, :file => "execution.jl", :line => 99, :path => "/home/runner/.julia/packages/BenchmarkTools/SnZV6/src/execution.jl"), Dict(:call => "#invokelatest#2", :inlined => true, :file => "essentials.jl", :line => 718, :path => "./essentials.jl"), Dict(:call => "#run_result#45", :inlined => true, :file => "execution.jl", :line => 34, :path => "/home/runner/.julia/packages/BenchmarkTools/SnZV6/src/execution.jl"), Dict(:call => "var\"#run#49\"(::Nothing, ::Float64, ::Float64, ::Base.Pairs{Symbol, Integer, NTuple{5, Symbol}, NamedTuple{(:verbose, :samples, :evals, :gctrial, :gcsample), Tuple{Bool, Int64, Int64, Bool, Bool}}}, ::typeof(run), ::BenchmarkTools.Benchmark, ::BenchmarkTools.Parameters)", :inlined => false, :file => "execution.jl", :line => 117, :path => "/home/runner/.julia/packages/BenchmarkTools/SnZV6/src/execution.jl"), Dict(:call => "#warmup#54", :inlined => true, :file => "execution.jl", :line => 169, :path => "/home/runner/.julia/packages/BenchmarkTools/SnZV6/src/execution.jl"), Dict(:call => "warmup(::BenchmarkTools.Benchmark)", :inlined => false, :file => "execution.jl", :line => 169, :path => "/home/runner/.julia/packages/BenchmarkTools/SnZV6/src/execution.jl"), Dict(:call => "macro expansion", :inlined => true, :file => "execution.jl", :line => 393, :path => "/home/runner/.julia/packages/BenchmarkTools/SnZV6/src/execution.jl"), Dict(:call => "top-level scope", :inlined => true, :file => "Wordlebase3.jl#==#6206bdbf-67e2-4469-98dc-3e62b75de93d", :line => 1, :path => "/home/runner/work/Wordlebase3/Wordlebase3/tutorials/Wordlebase3.jl#==#6206bdbf-67e2-4469-98dc-3e62b75de93d")]
```

**Comment by rikhuijzer** [comment]
Hmm there is something sketchy going on here still. After "fixing" the notebook and restarting the Pluto notebook, it reported an error again.

This seems to be a problem with the cell evaluation order from Pluto unfortunately :\

**Comment by rikhuijzer** [comment]
Got a MWE and am talking already with a Pluto developer at the Zulip chat:

```julia
using BenchmarkTools
```

```julia
@benchmark sum(x)
```

```julia
x = [1, 2]
```

This MWE fails when opening Pluto and succeeds when manually running the benchmark cell.

**Comment by rikhuijzer** [comment]
So the problem is already known https://github.com/fonsp/Pluto.jl/issues/1664

In Pluto.jl, it is not a huge problem, but for PlutoStaticHTML it is a shop stopper. As a hack, I could do a second pass over the notebook in PlutoStaticHTML if there are `@benchmark` cells.

**Comment by rikhuijzer** [comment]
I just released a patch/hack in PlutoStaticHTML version 3.2.2. Triggering CI once the new version is merged into the general registry should result in website updates again on your side :partying_face: Don't forget to set the Franklin page `title` as was also done in the two example notebooks; see the stuff in between the `+++`.

There is only one minor issue with the table generated by `playWordle(Base.Fix2(score, rand(tuples)), tuples)` still:

![image](https://user-images.githubusercontent.com/20724914/152542783-bc840da0-bace-4a8c-9c10-1e128a1d66fe.png)

This is caused PlutoStaticHTML converting complex Pluto (Javascript) objects back to static HTML objects. It is a known problem that very complex objects do not always get converted back properly (https://github.com/rikhuijzer/PlutoStaticHTML.jl/issues/53). There are various solutions to this problem:

1. Rewrite code on your side a bit so that outputs become easier to convert back (not ideal, but should work)
2. Fix this specific problem in PlutoStaticHTML (should be reasonably easy)
3. Disable the complex objects completely as mentioned in https://github.com/rikhuijzer/PlutoStaticHTML.jl/issues/53. I've already worked a bit on this, but it is made a bit difficult due to Pluto spawning a separate process which causes overrides by PlutoStaticHTML to be ignored.

PRs for fixing this on the PlutoStaticHTML side are welcome. Otherwise, it might take two weeks or so for a fix to be released. I'll have to let this problem sit in the back of my mind for a bit

Apart from this issue which occurs for one output, all looks good

**Comment by rikhuijzer** [comment]
@dmbates, all problems in the Wordle notebook have now been fixed in the just released PlutoStaticHTML 3.3.0. Specifically, the DataFrame with embedded tuples is now converted to HTML properly and `@benchmark` runs properly since PlutoStaticHTML 3.2.2.

Also, I saw that there was a CSS problem for the code blocks inside the Markdown. Add `display: block` to the `pre code` to fix that; see also https://github.com/rikhuijzer/JuliaTutorialsTemplate/commit/694f6aed9d69bce4fd3319580e60ff16d609686b.

Thanks for reporting an issue with an example notebook.
